std/range/package.d

```diff
@@ -5898,6 +5898,63 @@ if (!isIntegral!(CommonType!(B, E)) &&
     return Result(begin, end);
 }
 
+/// ditto
+auto iota(B, E, S)(B begin, E end, S step)
+if (!isIntegral!(CommonType!(B, E)) &&
+    !isFloatingPoint!(CommonType!(B, E)) &&
+    !isPointer!(CommonType!(B, E)) &&
+    is(typeof((ref B b, S s) { b += s; })) &&
+    is(typeof(S.init == 0)) &&
+    is(typeof(S.init < 0)) &&
+    is(typeof(B.init < E.init)))
```

Comment:
Could we unite the overloads, have only one function and use static if internally so that a user isn't hit by multiple giant blocks?

Reply:
Yes, that would be great. Just one function that accepts the union of all cases, and distinguish inside as needed using `static if` please.

Reply:
Yes, please reduce unnecessary user-facing overloads, unify into an API that makes sense from the user's POV.  Internally, either use static ifs inside the function body, or module-private overloads (e.g., `iotaImpl`) for dispatching to different specializations.